src/BlockTypes/MiniCart.php

```diff
@@ -183,6 +185,25 @@ protected function enqueue_data( array $attributes = [] ) {
 			'translations' => $this->get_inner_blocks_translations(),
 		);
 
+		// Preload inner blocks frontend scripts.
+		$inner_blocks_frontend_scripts = $cart->is_empty() ? array(
```

Comment:
`$cart` can potentially be null. So you might get a fatal in certain situations such as going to the theme->editor that has the mini cart block.

Reply:
Oh, good point. I refactored this in 9d08e0d6fd26f39dcd6de6a9f14fe77195c0a27b. I also noticed that we were calling `CartController`, but if I'm not wrong that was only intended to be used in Rest API calls, so I replaced those calls with calls to the `WC_Cart` instance.

In some of them I didn't add a check for `$cart` not being null because they are executed [after this check](https://github.com/woocommerce/woocommerce-blocks/blob/cb8b1ed296b8c45ef37f8648b092f5c40a491d9e/src/BlockTypes/MiniCart.php#L357). But maybe it would be better to be extra-cautious and make sure `$cart` is not null? What do you think?